azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/ContainerService.java

```diff
@@ -0,0 +1,263 @@
+            ContainerService.DefinitionStages.WithMasterProfile,
+            ContainerService.DefinitionStages.WithLinuxProfile,
+            ContainerService.DefinitionStages.WithLinuxProfileRootUsername,
+            ContainerService.DefinitionStages.WithLinuxProfileSshKey,
+            ContainerService.DefinitionStages.DefineAgentPoolProfiles,
+            ContainerService.DefinitionStages.WithCreate {
+    }
+
+    /**
+     * Grouping of container service definition stages.
+     */
+    interface DefinitionStages {
+        /**
+         * The first stage of a container service definition.
+         */
+        interface Blank extends
+                GroupableResource.DefinitionWithRegion<WithGroup> {
+        }
+
+        /**
+         * The stage of the container service definition allowing to specify the resource group.
+         */
+        interface WithGroup extends
+                GroupableResource.DefinitionStages.WithGroup<WithMasterProfile> {
+        }
+
+        /**
+         * The stage of the container service definition allowing to specify orchestration type.
+         */
+        interface WithOrchestratorProfile {
+            /**
+             * Specifies the Swarm orchestration type for the container service.
+             * @return the next stage of the definition
+             */
+            WithCreate withSwarmOrchestration();
+
+            /**
+             * Specifies the DCOS orchestration type for the container service.
+             * @return the next stage of the definition
+             */
+            WithCreate withDCOSOrchestration();
+
+            /**
+             * Specifies the Kubernetes orchestration type for the container service.
+             * @return the next stage of the definition
+             */
+            WithCreate withKubernetesOrchestration();
+        }
+
+        /**
+         * The stage of the container service definition allowing to specify the master profile.
+         */
+        interface WithMasterProfile {
+            /**
+             * The properties for master agents.
+             * @param count master profile count (1, 3, 5)
+             * @param dnsLabel the dns prefix
+             * @return the next stage of the definition
+             */
+            WithLinuxProfile withMasterProfile(ContainerServiceMasterProfileCount count, String dnsLabel);
+        }
+
+        /**
+         * The stage of the container service definition allowing to specify an agent pool profile.
+         */
+        interface DefineAgentPoolProfiles {
+            /**
+             * Begins the definition of a agent pool profile to be attached to the container service.
+             *
+             * @param name the name for the agent pool profile
+             * @return the stage representing configuration for the agent pool profile
+             */
+            CSAgentPoolProfile.DefinitionStages.Blank<WithCreate> defineContainerServiceAgentPoolProfile(String name);

```

Comment:
is the word 'Profile' at the end helpful here?  Would anything be lost here if this was just `defineContainerServiceAgentPool()`?  Or maybe even `defineAgentPool()` since it feels pretty obvious that we're in the Container Services context here?